scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -1239,6 +1262,50 @@ public final void handshakeFailed(Throwable cause) {
 		SecretUtil.destroy(this);
 	}
 
+	/**
+	 * Abort handshake.
+	 * 
+	 * Notifies all registered session listeners about a handshake failure.
+	 * Listeners are intended to keep the connection.
+	 * 
+	 * If {@link #setFailureCause(Throwable)} was called before, only calls with
+	 * the same cause will notify the listeners. If
+	 * {@link #setFailureCause(Throwable)} wasn't called before, sets the
+	 * <em>cause</em> property to the given cause.
+	 * 
+	 * @param cause The reason for the abort.
+	 * @see #handshakeFailed(Throwable)
+	 * @see #isRemovingConnection()
+	 */
+	public final void handshakeAborted(Throwable cause) {
+		this.handshakeAborted = true;
+		handshakeFailed(cause);
+	}
+
+	/**
+	 * Test, if handshake is expired according nano realtime.
+	 * 
+	 * Used to mitigate deep sleep during handhsakes.
```

Comment:
Maybe ? (personally I didn't know deep sleep before those PRs)

`Used to mitigate **Android** deep sleep during handhsakes`